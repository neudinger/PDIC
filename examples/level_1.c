/*
 * File: level_1.c
 * Project: examples
 * File Created: Friday, 23rd July 2021 12:48:58 am
 * Author: kbarre (kevin.barre@epitech.eu)
 * -----
 * Last Modified: Friday, 23rd July 2021 12:49:20 am
 * Modified By: kbarre (kevin.barre@epitech.eu>)
 * -----
 * Licenses: EUPL
 * -----
 * Copyright 2021 - 2021 NinjaTech
 */

short notsee;

#pragma pdi on

int global_int;
char global_char;
void *global_pointer;
float **global_float_pointer_of_pointer;
int array[24];

// pointer to array of int of size 42
#pragma pdi type:int32; size:[42]
int **pointer_of_array;

double *array_of_pointers[21];

#pragma pdi type:uint64
#pragma pdi size:[10][10][10]
unsigned ****my_cube;

#pragma pdi off

unsigned short notsee_either;