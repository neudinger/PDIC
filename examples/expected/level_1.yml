data:
 array: { type: array, subtype: int, size: 24 }
 array_of_pointers: { type: array, subtype:  { type: pointer, subtype: double }, size: 21 }
 global_char: 
  type: char
 global_float_pointer_of_pointer: { type: pointer, subtype:  { type: pointer, subtype: float } }
 global_int: 
  type: int
 global_pointer: { type: pointer, subtype: void }
 my_cube: { type: pointer, subtype:  { type: array, subtype: uint64, size: [10, 10, 10] } }
 pointer_of_array: { type: pointer, subtype:  { type: array, subtype: int32, size: 42 } }
